docs/Toy Tutorial/examples/codegen.toy

```diff
@@ -0,0 +1,31 @@
+# RUN: toyc-ch2 %s -emit=mlir 2>&1 | FileCheck %s
```

Comment:
This is already chapter 2, right? Is there a reason why this is contained in this PR?

Reply:
I added all the toy examples, and didn't bother removing any of the comments, it's not like we're running filecheck on them. Happy to remove the comments if you think they might be confusing

Reply:
But we should right? Can we?

Reply:
We technically can't right now because we don't match the MLIR syntax exactly, even in MLIR mode on the printer. I think it would be a nice goal, but a bit orthogonal to this PR